client/showid.lua

```diff
@@ -0,0 +1,55 @@
+function DrawPlayerName(x, y, z, name)
+    local onScreen, _x, _y = GetScreenCoordFromWorldCoord(x, y, z)
+    local px, py, pz = table.unpack(GetFinalRenderedCamCoord())
+    local dist = #(vector3(px, py, pz) - vector3(x, y, z))
+
+    local scale = ((1/dist)*2)*((1/GetGameplayCamFov())*100)
+
+    if onScreen then
+        SetTextScale(0.0*scale, 0.5*scale)
+        SetTextFont(0)
+        SetTextProportional(1)
+        SetTextColour(255, 255, 255, 255)
+        SetTextDropshadow(0, 0, 0, 0, 255)
+        SetTextEdge(2, 0, 0, 0, 150)
+        SetTextDropShadow()
+        SetTextOutline()
+        BeginTextCommandDisplayText("STRING")
+        SetTextCentre(1)
+        AddTextComponentSubstringPlayerName(name)
+		GetScreenCoordFromWorldCoord(x, y, z, 0)
+        EndTextCommandDisplayText(_x, _y)
+    end
+end
+
+local showNames = false
+
+CreateThread(function()
+    while true do
+        local sleep = 500
+        if showNames then
+            sleep = 0
+            for _, id in ipairs(GetActivePlayers()) do
+                if NetworkIsPlayerActive(id) then
+                    local myPos = GetEntityCoords(PlayerPedId())
+                    local otherPos = GetEntityCoords(GetPlayerPed(id))
+                    local distance = #(myPos - otherPos)
+                    if distance < 15.0 then
+                        DrawPlayerName(otherPos.x, otherPos.y, otherPos.z + 1.0, GetPlayerServerId(id))
+                    end
+                end
+            end
+        end
+        Wait(sleep)
+    end
+end)
+
+RegisterCommand('+showid', function()
+    showNames = true
+end, false)
+
+RegisterCommand('-showid', function()
+    showNames = false
+end, false)
+
+RegisterKeyMapping('+showid', 'Show ID\'s', 'keyboard', 'EQUALS')
```

Comment:
Why is seeing other player's ids a desired feature? Seems like an easy way to meta-game players vs AI

Reply:
Yeah this is not needed, the scoreboard already has this integrated with keymapping

Reply:
To report players, you can only view it if you hold it down and once you release the key it will go away. It's hard to report people when you don't know their ID.

Reply:
But it's in qb-scoreboard

Reply:
Yup, qb-scoreboard does the exact same, if not in a better way

Reply:
Ah okay removed